index.html

```diff
@@ -0,0 +1,125 @@
+    <meta name="msapplication-TileColor" content="#3fbbf4ff"/>
+    <meta name='mobile-web-app-capable' content='yes'>
+    <meta name='viewport' content='width=device-width, user-scalable=no'>
+    <meta name='theme-color' content='{{ theme_color }}'>
+    <style>
+      body {
+        font-family: 'Roboto', 'Noto', sans-serif;
+        font-weight: 400;
+        -webkit-font-smoothing: antialiased;
+        text-rendering: optimizeLegibility;
+        margin: 0;
+        padding: 0;
+      }
+
+      #ha-init-skeleton::before {
+        display: block;
+        content: "";
+        height: 48px;
+        background-color: {{ theme_color }};
+      }
+
+      #ha-init-skeleton .message {
+        transition: font-size 2s;
+        font-size: 0;
+        padding: 24px;
+      }
+
+      #ha-init-skeleton.error .message {
+        font-size: 16px;
+      }
+
+      #ha-init-skeleton a {
+        color: {{ theme_color }};
+        text-decoration: none;
+        font-weight: bold;
+      }
+    </style>
+    <script>
+      function initError() {
+        document.getElementById('ha-init-skeleton').classList.add('error');
+      };
+      window.noAuth = {{ no_auth }};
+      window.Polymer = {
+        lazyRegister: true,
+        useNativeCSSProperties: true,
+        dom: 'shadow',
+        suppressTemplateNotifications: true,
+        suppressBindingNotifications: true,
+      };
+    </script>
+  </head>
+  <body>
+    <div id='ha-init-skeleton'>
+      <div class='message'>
+        Home Assistant had trouble<br>connecting to the server.<br><br>
+        <a href='/'>TRY AGAIN</a>
+      </div>
+    </div>
+    <home-assistant></home-assistant>
+    {# <script src='/home-assistant-polymer/build/_demo_data_compiled.js'></script> -#}
+    {% if not latest -%}
+    <script>
+      var compatibilityRequired = (typeof Object.assign != 'function');
+      if (compatibilityRequired) {
+        var e = document.createElement('script');
+        e.onerror = initError;
+        e.src = '/home-assistant-polymer/build-temp/compatibility.js';
+        document.head.appendChild(e);
+      }
+    </script>
+    {% endif -%}
+    <script src='/home-assistant-polymer/build/core.js'></script>
+    {% if not dev_mode and not latest -%}
+    <script src='/frontend_es5/custom-elements-es5-adapter.js'></script>
+    {% endif -%}
+    <script>
+      var webComponentsSupported = (
+        'customElements' in window &&
+        'import' in document.createElement('link') &&
+        'content' in document.createElement('template'));
+      if (!webComponentsSupported) {
+        var e = document.createElement('script');
+        e.onerror = initError;
+        e.src = '/static/webcomponents-lite.js';
+        document.head.appendChild(e);
+      }
+      if ('serviceWorker' in navigator) {
+        window.addEventListener('load', function () {
+          navigator.serviceWorker.register(
+            '/home-assistant-polymer/build/service_worker.js');
```

Comment:
service_worker's scope is only under its own dir, so it must be in root.
So I think service worker code should be kept in python (it is not fingerprinted anyway)

Reply:
Fixed.